selftests/functional/test_statuses.py

```diff
@@ -0,0 +1,206 @@
+                                         'teardown pre',
+                                         'teardown post']),
+                    'exit-test-f680': ('ERROR',
+                                       ['setup pre',
+                                        'setup post',
+                                        'test pre',
+                                        'teardown pre',
+                                        'teardown post']),
+                    'exit-teardown-6304': ('ERROR',
+                                           ['setup pre',
+                                            'setup post',
+                                            'test pre',
+                                            'test post',
+                                            'teardown pre']),
+                    'exception-setup-37d3': ('ERROR',
+                                             ['setup pre',
+                                              'teardown pre',
+                                              'teardown post']),
+                    'exception-test-8f7d': ('ERROR',
+                                            ['setup pre',
+                                             'setup post',
+                                             'test pre',
+                                             'teardown pre',
+                                             'teardown post']),
+                    'exception-teardown-dfb1': ('ERROR',
+                                                ['setup pre',
+                                                 'setup post',
+                                                 'test pre',
+                                                 'test post',
+                                                 'teardown pre']),
+                    'kill-test-57fe': ('ERROR',
+                                       ['setup pre',
+                                        'setup post',
+                                        'test pre']),
+                    }
+
+
+class TestStatuses(unittest.TestCase):
+
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp(prefix='avocado_' + __name__)
+        test_file = os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                                 os.path.pardir,
+                                                 ".data",
+                                                 'test_statuses.py'))
+        yaml_file = os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                                 os.path.pardir,
+                                                 ".data",
+                                                 'test_statuses.yaml'))
+
+        cmd = ('%s run %s -m %s --sysinfo=off --job-results-dir %s --json -' %
+               (AVOCADO, test_file, yaml_file, self.tmpdir))
+
+        results = process.run(cmd, ignore_status=True)
+        self.results = json.loads(results.stdout)
```

Comment:
This is not a test. In case this fails, we have to inspect and fix our selftests, not our code. If you want extra debug info here, we have a  massive work ahead to make such error handling in almost every single selftests we have.